caffe2/mkl/operators/pool_op.cc

```diff
@@ -0,0 +1,196 @@
+#include "caffe2/core/context.h"
+#include "caffe2/core/operator.h"
+#include "caffe2/operators/conv_pool_op_base.h"
+
+#include "caffe2/utils/mkl_utils.h"
+
+#ifdef CAFFE2_HAS_MKL_DNN
+
+namespace caffe2 {
+
+namespace {
+// These two classe are just used as template arguments passed to the PoolOp
+// template to instantiate the different algorithms.
+class AveragePool {};
+class MaxPool {};
+}  // namespace
+
+
+namespace mkl {
+
+template <typename T, typename PoolType>
+class MKLPoolOp final : public ConvPoolOpBase<MKLContext>{
+
+  public:
+    USE_CONV_POOL_BASE_FUNCTIONS(MKLContext);
+    
+    MKLPoolOp(const OperatorDef &operator_def, Workspace *ws)
+        : ConvPoolOpBase<MKLContext>(operator_def, ws) {
+
+      CAFFE_ENFORCE(
+            dilation_h_ == 1 && dilation_w_ == 1,
+            "Pooling op does not support dilation right now.");
+        if (!global_pooling_) {
+      CAFFE_ENFORCE(
+          pad_t_ < kernel_h_ && pad_b_ < kernel_h_ && pad_l_ < kernel_w_ &&
+              pad_r_ < kernel_w_,
+          "Pad should be smaller than kernel.");
+        }
+    }
+    
+    ~MKLPoolOp() {
+        if (workspace_buffer_ != NULL) {
```

Comment:
Ah got it - let me check the linting situation and come back to you.

Regarding testing, could you also add a testing script to cover the numerical correctness? Something like this:

https://github.com/caffe2/caffe2/blob/master/caffe2/python/operator_test/mkl_conv_op_test.py

and we can use the scaffolding from the existing pooling op tests:

https://github.com/caffe2/caffe2/blob/master/caffe2/python/operator_test/pooling_test.py